### ISSUE TITLE
Warn when calling ReactDOM.render without global "document", etc available

### ISSUE THREAD

**Issue by dmatteo** [facebook/react#4881]
Basically, every time `setState` is called and a rerender happens, I get this error:  
```
Invariant Violation: dangerouslyRenderMarkup(...): Cannot render markup in a worker thread. 
Make sure `window` and `document` are available globally before requiring React 
when unit testing or use React.renderToString for server rendering.
```

I've created a very bare repo where you can recreate the issue:
https://github.com/dmatteo/setStateBoom

The component + test suite is so small that I can copy/paste it here:

```javascript

import React, {Component} from 'react/addons';
import jsdomify from 'jsdomify';
import expect from 'unexpected';

class MyComponent extends Component {
  constructor(props) {
    super(props);
    this.displayName = 'MyComponent';
    this.state = {
      bananas: 'bananas'
    };
  }

  render() {
    return <div>{this.state.bananas}</div>
  }
}

describe('setState test', () => {

  before(() => {
    // this leaks document and window to global
    jsdomify.create();
  });

  it('should render', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    expect(instance, 'to be defined');
  });

  it('should not throw', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    let foo = () => {
      instance.setState({bananas: 'apples'});
    };
    expect(foo, 'not to throw');
  })
  
});
```

Do you have any idea what is happening and what can I do about it? (if anything)

**Comment by edvinerikson** [comment]
It's probably because you import React before you setup jsdom.
http://stackoverflow.com/a/26872245

**Comment by dmatteo** [comment]
It's not complaining about a missing document when I import react and the first test pass without issues.

**Comment by edvinerikson** [comment]
This code is working for me:
```javascript
import expect from 'unexpected';
require('jsdomify').create();
var React = require('react/addons');
var { Component } = React;

class MyComponent extends Component {
  constructor(props) {
    super(props);
    this.displayName = 'MyComponent';
    this.state = {
      bananas: 'bananas'
    };
  }

  render() {
    return <div>{this.state.bananas}</div>
  }
}

describe('setState test', () => {

  before(() => {
    //jsdomify.create();
  });

  it('should render', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    expect(instance, 'to be defined');
  });

  it('should not throw', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    let foo = () => {
      instance.setState({bananas: 'apples'});
    };
    expect(foo, 'not to throw');
  })


});
```

**Comment by dmatteo** [comment]
That's very interesting.
Also this works:
```
import expect from 'unexpected';
import jsdomify from 'jsdomify;
jsdomify.create()
var React = require('react/addons');
var { Component } = React;
```

But this **doesn't** (although it should, I guess)
```
import expect from 'unexpected';
import jsdomify from 'jsdomify;
jsdomify.create()
import React, {Component} from 'react/addons'
```

I suspect hoisting might be the problem, but if this is the case the working solutions feels very brittle.

**Comment by dmatteo** [comment]
Still, in my real scenario, I still have the error, even with your suggestion implemented.
```
Error: Invariant Violation: dangerouslyRenderMarkup(...): Cannot render markup in a 
worker thread. Make sure `window` and `document` are available globally before
 requiring React when unit testing or use React.renderToString for server rendering.
```
  
  
I will try to (re)reproduce this in my reduce test example

**Comment by edvinerikson** [comment]
I updated my example to be more clear.
Note the `require('react/lib/ExecutionEnvironment').canUseDOM = true;` in the `before` hook.
```javascript
import React, {Component} from 'react/addons';
import jsdomify from 'jsdomify';
import expect from 'unexpected';

class MyComponent extends Component {
  constructor(props) {
    super(props);
    this.displayName = 'MyComponent';
    this.state = {
      bananas: 'bananas'
    };
  }

  render() {
    return <div>{this.state.bananas}</div>
  }
}

describe('setState test', () => {
  before(() => {
    jsdomify.create();
    require('react/lib/ExecutionEnvironment').canUseDOM = true;
  });

  it('should render', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    expect(instance, 'to be defined');
  });

  it('should not throw', () => {
    let instance = React.addons.TestUtils.renderIntoDocument(<MyComponent></MyComponent>);
    let foo = () => {
      instance.setState({bananas: 'apples'});
    };
    expect(foo, 'not to throw');
  })
});
```

**Comment by edvinerikson** [comment]
Anyway this is kind of a duplicate of https://github.com/facebook/react/issues/3213, https://github.com/facebook/react/issues/3840, https://github.com/facebook/react/issues/2019 and is already being tracked in https://github.com/facebook/react/issues/4592.

Maybe we should write a note in http://facebook.github.io/react/docs/test-utils.html telling you that you need `window`, `document` and `document.createElement` globally available before importing React.

**Comment by spicyj** [comment]
Yeah, perhaps we can add a warning for this (until we fix #4592).
